baseapp/abci.go

```diff
@@ -304,7 +304,7 @@ func (app *BaseApp) VerifyFraudProof(req abci.RequestVerifyFraudProof) (res abci
 		}
 
 		appHash = appFromFraudProof.GetAppHash(abci.RequestGetAppHash{}).AppHash
-		success = bytes.Equal(appHash, req.ExpectedAppHash)
+		success = bytes.Equal(appHash, req.ExpectedValidAppHash)
```

Comment:
`VerifyFraudProof` should return `true` if and only if fraud proof is completely valid and proves that fraudulent state transition happened.
`appHash` is generated locally, so we must trust it.
`req.ExpectedValidAppHash` was received from P2P network, so we're not sure it's correct. By comparing here we ensure that received fraud proof was honest.

Reply:
Ensuring that block actually contained different (invalid) ISR/appHash, is a separate issue - do we have one in GitHub?

Reply:
Just made one here: https://github.com/celestiaorg/cosmos-sdk-rollmint/issues/12